README.md
tekticles is a simple set of fonts, derived from ClearFont2. I only ever used
one set of fonts from CF2 (the Calibri set), so I ripped CF2 down to it's bear
ass and stripped off all the extra crap.  This addon is what's left.


changelog.txt
6.0.0.18-Beta
    6.0 updates (thanks @arcadepro)

5.4.0.17-Beta
    Fix nameplates for 5.4

5.1.0.16-Release
    It be stable or something

5.0.1.15-Beta
    New fonts
    Improve the background of the config page

5.0.1.14-Beta
    Seems we have some new fonts in game that need to be set

4.2.0.13-Beta
    2 column config panel now, woo!
    Remove delayed load, we really do want the fonts applied at load

4.1.0.12-Beta
    Fix channel list buttons not getting fonts
    Fix title picker buttons in 4.1

3.3.0.11-Release
    Seems that the friends frame defined its own fonts, ugh

3.2.0.10-Release
    I can't, for the life of me, find where SetFont is being called to break font inheritance... so fuck it I'll just force the font objects back to what they're supposed to be.

3.0.9.9-Release
    Forgot libstub

3.0.3.8-Beta
    ReputationDetailFont doesn't inherit anymore, ugheth

3.0.3.7-Beta
    Missed SystemFont_Shadow_Huge3
    Missed SystemFont_Small

3.0.2.6-Beta
    Convert everything to Blizzy's new base fonts.  I don't know why they made these... meh
    Add about panel
    Add preview panel

3.0.1.5-Beta
    Remove fonts that are gone in Wrath

2.4.2.4-Beta
    Fix for people who don't have LibStub (that's possible?)

2.4.2.3-Beta
    Apply font to BossEmoteNormalHuge

2.4.1.2-Beta
    Register fonts with LMS3, if present

2.4.1.1-Beta
    Fix them thar font paths
    And so it begins...


tekticles.lua

local function SetFont(obj, font, size, style, r, g, b, sr, sg, sb, sox, soy)
	obj:SetFont(font, size, style)
	if sr and sg and sb then obj:SetShadowColor(sr, sg, sb) end
	if sox and soy then obj:SetShadowOffset(sox, soy) end
	if r and g and b then obj:SetTextColor(r, g, b)
	elseif r then obj:SetAlpha(r) end
end


local f = CreateFrame("Frame")
f:RegisterEvent("ADDON_LOADED")
f:SetScript("OnEvent", function()
	local NORMAL     = "Interface\\AddOns\\tekticles\\Calibri.ttf"
	local BOLD       = "Interface\\AddOns\\tekticles\\CalibriBold.ttf"
	local BOLDITALIC = "Interface\\AddOns\\tekticles\\CalibriBoldItalic.ttf"
	local ITALIC     = "Interface\\AddOns\\tekticles\\CalibriItalic.ttf"
	local NUMBER     = "Interface\\AddOns\\tekticles\\CalibriBold.ttf"

	UIDROPDOWNMENU_DEFAULT_TEXT_HEIGHT = 12
	CHAT_FONT_HEIGHTS = {7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24}

	UNIT_NAME_FONT     = NORMAL
	DAMAGE_TEXT_FONT   = NUMBER
	STANDARD_TEXT_FONT = NORMAL

	-- Base fonts
	SetFont(AchievementFont_Small,                BOLD, 12)
	SetFont(FriendsFont_Large,                  NORMAL, 15, nil, nil, nil, nil, 0, 0, 0, 1, -1)
	SetFont(FriendsFont_Normal,                 NORMAL, 13, nil, nil, nil, nil, 0, 0, 0, 1, -1)
	SetFont(FriendsFont_Small,                  NORMAL, 11, nil, nil, nil, nil, 0, 0, 0, 1, -1)
	SetFont(FriendsFont_UserText,               NUMBER, 12, nil, nil, nil, nil, 0, 0, 0, 1, -1)
	SetFont(GameTooltipHeader,                    BOLD, 15, "OUTLINE")
	SetFont(GameFont_Gigantic,                    BOLD, 32, nil, nil, nil, nil, 0, 0, 0, 1, -1)
	SetFont(GameNormalNumberFont,                 BOLD, 11)
	SetFont(InvoiceFont_Med,                    ITALIC, 13, nil, 0.15, 0.09, 0.04)
	SetFont(InvoiceFont_Small,                  ITALIC, 11, nil, 0.15, 0.09, 0.04)
	SetFont(MailFont_Large,                     ITALIC, 15, nil, 0.15, 0.09, 0.04, 0.54, 0.4, 0.1, 1, -1)
	SetFont(NumberFont_OutlineThick_Mono_Small, NUMBER, 13, "OUTLINE")
	SetFont(NumberFont_Outline_Huge,            NUMBER, 30, "THICKOUTLINE", 30)
	SetFont(NumberFont_Outline_Large,           NUMBER, 17, "OUTLINE")
	SetFont(NumberFont_Outline_Med,             NUMBER, 15, "OUTLINE")
	SetFont(NumberFont_Shadow_Med,              NORMAL, 14)
	SetFont(NumberFont_Shadow_Small,            NORMAL, 12)
	SetFont(QuestFont_Shadow_Small,             NORMAL, 16)
	SetFont(QuestFont_Large,                    NORMAL, 16)
	SetFont(QuestFont_Shadow_Huge,                BOLD, 19, nil, nil, nil, nil, 0.54, 0.4, 0.1)
	SetFont(QuestFont_Super_Huge,                 BOLD, 24)
	SetFont(ReputationDetailFont,                 BOLD, 12, nil, nil, nil, nil, 0, 0, 0, 1, -1)
	SetFont(SpellFont_Small,                      BOLD, 11)
	SetFont(SystemFont_InverseShadow_Small,       BOLD, 11)
	SetFont(SystemFont_Large,                   NORMAL, 17)
	SetFont(SystemFont_Med1,                    NORMAL, 13)
	SetFont(SystemFont_Med2,                    ITALIC, 14, nil, 0.15, 0.09, 0.04)
	SetFont(SystemFont_Med3,                    NORMAL, 15)
	SetFont(SystemFont_OutlineThick_Huge2,      NORMAL, 22, "THICKOUTLINE")
	SetFont(SystemFont_OutlineThick_Huge4,  BOLDITALIC, 27, "THICKOUTLINE")
	SetFont(SystemFont_OutlineThick_WTF,    BOLDITALIC, 31, "THICKOUTLINE", nil, nil, nil, 0, 0, 0, 1, -1)
	SetFont(SystemFont_Outline_Small,           NUMBER, 13, "OUTLINE")
	SetFont(SystemFont_Shadow_Huge1,              BOLD, 20)
	SetFont(SystemFont_Shadow_Huge3,              BOLD, 25)
	SetFont(SystemFont_Shadow_Large,            NORMAL, 17)
	SetFont(SystemFont_Shadow_Med1,             NORMAL, 13)
	SetFont(SystemFont_Shadow_Med2,             NORMAL, 14)
	SetFont(SystemFont_Shadow_Med3,             NORMAL, 15)
	SetFont(SystemFont_Shadow_Outline_Huge2,    NORMAL, 22, "OUTLINE")
	SetFont(SystemFont_Shadow_Small,              BOLD, 11)
	SetFont(SystemFont_Small,                   NORMAL, 12)
	SetFont(SystemFont_Tiny,                    NORMAL, 11)
	SetFont(Tooltip_Med,                        NORMAL, 13)
	SetFont(Tooltip_Small,                        BOLD, 12)
	SetFont(WhiteNormalNumberFont,                BOLD, 11)

	-- Derived fonts
	SetFont(BossEmoteNormalHuge,     BOLDITALIC, 27, "THICKOUTLINE")
	SetFont(CombatTextFont,              NORMAL, 26)
	SetFont(ErrorFont,                   ITALIC, 16, nil, 60)
	SetFont(QuestFontNormalSmall,          BOLD, 13, nil, nil, nil, nil, 0.54, 0.4, 0.1)
	SetFont(WorldMapTextFont,        BOLDITALIC, 31, "THICKOUTLINE",  40, nil, nil, 0, 0, 0, 1, -1)

	for i=1,7 do
		local f = _G["ChatFrame"..i]
		local font, size = f:GetFont()
		f:SetFont(NORMAL, size)
	end

	-- I have no idea why the channel list is getting fucked up
	-- but re-setting the font obj seems to fix it
	for i=1,MAX_CHANNEL_BUTTONS do
		local f = _G["ChannelButton"..i.."Text"]
		f:SetFontObject(GameFontNormalSmallLeft)
		-- function f:SetFont(...) error("Attempt to set font on ChannelButton"..i) end
	end

	for _,butt in pairs(PaperDollTitlesPane.buttons) do butt.text:SetFontObject(GameFontHighlightSmallLeft) end
end)


local LSM = LibStub and LibStub:GetLibrary("LibSharedMedia-3.0", true)
if LSM then
	LSM:Register(LSM.MediaType.FONT, "Calibri", "Interface\\AddOns\\tekticles\\Calibri.ttf")
	LSM:Register(LSM.MediaType.FONT, "Calibri Bold", "Interface\\AddOns\\tekticles\\CalibriBold.ttf")
	LSM:Register(LSM.MediaType.FONT, "Calibri Bold Italic", "Interface\\AddOns\\tekticles\\CalibriBoldItalic.ttf")
	LSM:Register(LSM.MediaType.FONT, "Calibri Italic", "Interface\\AddOns\\tekticles\\CalibriItalic.ttf")
end


tekKonfigScroll.lua

local lib, oldminor = LibStub:NewLibrary("tekKonfig-Scroll", 1)
if not lib then return end

lib.bg = {
	edgeFile = "Interface\\Tooltips\\UI-Tooltip-Border",
	tile = true,
	tileSize = 16,
	edgeSize = 12,
	insets = { left = 0, right = 0, top = 5, bottom = 5 }
}

-- Creates a scrollbar
-- Parent is required, offset is optional
function lib.new(parent, offset)
	local f = CreateFrame("Slider", nil, parent)
	f:SetWidth(16)

	f:SetPoint("TOPRIGHT", 0 - (offset or 0), -16 - (offset or 0))
	f:SetPoint("BOTTOMRIGHT", 0 - (offset or 0), 16 + (offset or 0))

	local up = CreateFrame("Button", nil, f)
	up:SetPoint("BOTTOM", f, "TOP")
	up:SetWidth(16) up:SetHeight(16)
	up:SetNormalTexture("Interface\\Buttons\\UI-ScrollBar-ScrollUpButton-Up")
	up:SetPushedTexture("Interface\\Buttons\\UI-ScrollBar-ScrollUpButton-Down")
	up:SetDisabledTexture("Interface\\Buttons\\UI-ScrollBar-ScrollUpButton-Disabled")
	up:SetHighlightTexture("Interface\\Buttons\\UI-ScrollBar-ScrollUpButton-Highlight")

	up:GetNormalTexture():SetTexCoord(1/4, 3/4, 1/4, 3/4)
	up:GetPushedTexture():SetTexCoord(1/4, 3/4, 1/4, 3/4)
	up:GetDisabledTexture():SetTexCoord(1/4, 3/4, 1/4, 3/4)
	up:GetHighlightTexture():SetTexCoord(1/4, 3/4, 1/4, 3/4)
	up:GetHighlightTexture():SetBlendMode("ADD")

	up:SetScript("OnClick", function(self)
		local parent = self:GetParent()
		parent:SetValue(parent:GetValue() - parent:GetHeight()/2)
		PlaySound("UChatScrollButton")
	end)

	local down = CreateFrame("Button", nil, f)
	down:SetPoint("TOP", f, "BOTTOM")
	down:SetWidth(16) down:SetHeight(16)
	down:SetNormalTexture("Interface\\Buttons\\UI-ScrollBar-ScrollDownButton-Up")
	down:SetPushedTexture("Interface\\Buttons\\UI-ScrollBar-ScrollDownButton-Down")
	down:SetDisabledTexture("Interface\\Buttons\\UI-ScrollBar-ScrollDownButton-Disabled")
	down:SetHighlightTexture("Interface\\Buttons\\UI-ScrollBar-ScrollDownButton-Highlight")

	down:GetNormalTexture():SetTexCoord(1/4, 3/4, 1/4, 3/4)
	down:GetPushedTexture():SetTexCoord(1/4, 3/4, 1/4, 3/4)
	down:GetDisabledTexture():SetTexCoord(1/4, 3/4, 1/4, 3/4)
	down:GetHighlightTexture():SetTexCoord(1/4, 3/4, 1/4, 3/4)
	down:GetHighlightTexture():SetBlendMode("ADD")

	down:SetScript("OnClick", function(self)
		local parent = self:GetParent()
		parent:SetValue(parent:GetValue() + parent:GetHeight()/2)
		PlaySound("UChatScrollButton")
	end)

	f:SetThumbTexture("Interface\\Buttons\\UI-ScrollBar-Knob")
	local thumb = f:GetThumbTexture()
	thumb:SetWidth(16) thumb:SetHeight(24)
	thumb:SetTexCoord(1/4, 3/4, 1/8, 7/8)

	f:SetScript("OnValueChanged", function(self, value)
		local min, max = self:GetMinMaxValues()
		if value == min then up:Disable() else up:Enable() end
		if value == max then down:Disable() else down:Enable() end
	end)

	local border = CreateFrame("Frame", nil, f)
	border:SetPoint("TOPLEFT", up, -5, 5)
	border:SetPoint("BOTTOMRIGHT", down, 5, -3)
	border:SetBackdrop(lib.bg)
	border:SetBackdropBorderColor(TOOLTIP_DEFAULT_COLOR.r, TOOLTIP_DEFAULT_COLOR.g, TOOLTIP_DEFAULT_COLOR.b, 0.5)

	return f, up, down, border
end


sample.lua

local frame = CreateFrame("ScrollFrame", nil, InterfaceOptionsFramePanelContainer)
frame.name = "tekticles"
frame:Hide()

frame:SetScript("OnShow", function(self)
	local title, subtitle = LibStub("tekKonfig-Heading").new(self, "tekticles", "This panel shows a preview for every major font template provided by Blizzard.")

	local group = LibStub("tekKonfig-Group").new(self)
	group:SetPoint("TOP", subtitle, "BOTTOM", 0, -8)
	group:SetPoint("BOTTOMLEFT", 16, 16)
	group:SetPoint("BOTTOMRIGHT", -16, 16)


	local tex = group:CreateTexture(nil, "BACKGROUND")
	tex:SetTexture("Interface\\FrameGeneral\\UI-Background-Rock")
	tex:SetPoint("TOPLEFT", 4, -4)
	tex:SetPoint("BOTTOMRIGHT", -4, 4)

	local alpha = group:CreateTexture(nil, "BACKGROUND")
	alpha:SetTexture(1, 1, 1, 0.5)
	alpha:SetBlendMode('ADD')
	alpha:SetAllPoints(tex)


	local GAP = 5
	local frame = CreateFrame("Frame", nil, group)
	frame:SetPoint("TOPRIGHT", -GAP, -GAP)
	frame:SetPoint("BOTTOMLEFT", GAP, GAP)

	local last
	local fonts, needbackground = {
		"GameFontNormal",
		"GameFontHighlight",
		"GameFontDisable",
		"GameFontNormalSmall",
		"WhiteNormalNumberFont",
		"GameNormalNumberFont",
		"GameFontHighlightExtraSmall",
		"GameFontHighlightMedium",
		"GameFontNormalLarge",
		"GameFontNormalHuge",
		"GameFont_Gigantic",
		"BossEmoteNormalHuge",
		"NumberFontNormal",
		"NumberFontNormalSmall",
		"NumberFontNormalLarge",
		"NumberFontNormalHuge",
		"ChatFontNormal",
		"ChatFontSmall",
		"DialogButtonNormalText",
		"ZoneTextFont",
		"SubZoneTextFont",
		"PVPInfoTextFont",
		"QuestFont_Super_Huge",
		"QuestFont_Shadow_Small",
		"ErrorFont",
		"TextStatusBarText",
		"CombatLogFont",
		"GameTooltipText",
		"GameTooltipTextSmall",
		"GameTooltipHeaderText",
		"WorldMapTextFont",
		"CombatTextFont",
		"MovieSubtitleFont",
		"AchievementPointsFont",
		"AchievementPointsFontSmall",
		"AchievementDescriptionFont",
		"AchievementCriteriaFont",
		"AchievementDateFont",
		"ReputationDetailFont",
		"QuestTitleFont",
		"QuestFont",
		"QuestFontNormalSmall",
		"QuestFontHighlight",
		"ItemTextFontNormal",
		"MailTextFontNormal",
		"SubSpellFont",
		"InvoiceTextFontNormal",
		"InvoiceTextFontSmall",
	}, {
		QuestFont = true,
		QuestFontNormalSmall = true,
		QuestTitleFont = true,
		QuestFontHighlight = true,
		ItemTextFontNormal = true,
		InvoiceTextFontNormal = true,
		InvoiceTextFontSmall = true,
		MailTextFontNormal = true,
		SubSpellFont = true,
	}
	for i,font in pairs(fonts) do
		local fs = frame:CreateFontString(nil, "ARTWORK", font)
		if i == 1 then
			fs:SetPoint("TOPLEFT", frame, "TOPLEFT", 30, -15)
		elseif i == 23 then
			fs:SetPoint("TOPLEFT", frame, "TOP", 30, -15)
		else
			fs:SetPoint("TOPLEFT", last, "BOTTOMLEFT", 0, -4)
		end
		fs:SetText(font)
		if needbackground[font] then
			local tex = frame:CreateTexture(nil, "ARTWORK")
			tex:SetPoint("TOPLEFT", fs, -2, 2)
			tex:SetPoint("BOTTOM", fs, 0, -2)
			tex:SetWidth(150)
			tex:SetTexture(1, 1, 1, 0.4)
			tex:SetBlendMode('ADD')
		end
		last = fs
	end

	self:SetScript("OnShow", nil)
end)

InterfaceOptions_AddCategory(frame)
